pkg/controller/nodes/task/plugin_config.go

```diff
@@ -13,23 +13,21 @@ import (
 	"github.com/flyteorg/flytepropeller/pkg/controller/nodes/task/k8s"
 )
 
-func WranglePluginsAndGenerateFinalList(ctx context.Context, cfg *config.TaskPluginConfig, pr PluginRegistryIface) ([]core.PluginEntry, error) {
+func WranglePluginsAndGenerateFinalList(ctx context.Context, cfg *config.TaskPluginConfig, pr PluginRegistryIface) (enabledPlugins []core.PluginEntry, defaultForTaskTypes map[pluginID][]taskType, err error) {
 	allPluginsEnabled := false
 	pluginsConfigMeta := config.PluginsConfigMeta{
-		AllDefaultForTaskTypes: map[pluginID][]taskType{},
+		AllDefaultForTaskTypes: defaultForTaskTypes,
 	}
-	var err error
 	if cfg != nil {
 		pluginsConfigMeta, err = cfg.GetEnabledPlugins()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
```

Comment:
I'm trying to read through this function and I think the logic here is a bit twisted (I understand it's not part of the PR but just trying to leave the code a bit better than when we started)... 
In what scenarios do we pass a nil `cfg` and why do we expect that to run through the function as opposed to erroring...

This block feels like it should be re-written to something like:
```golang
    if cfg == nil {
        return nil, nil, fmt.Errorf("cfg is a required argument")
    }

    pluginsConfigMeta, err := cfg.GetEnabledPlugins()
    if err != nil {
        return nil, nil, err
    }

Reply:
Since we default the TaskPlugins config to [empty lists](https://github.com/flyteorg/flytepropeller/blob/3baef11bb401c1bf7523e26d3423e356814e5e00/pkg/controller/nodes/task/config/config.go#L21) there is no place where we can get nil within FlytePropeller. I validated by passing empty configs on startup. I updated to error on a nil config.